File: android/src/main/java/one/tesseract/devwallet/rust/TestSettingsProvider.kt
package one.tesseract.devwallet.rust

import one.tesseract.devwallet.entity.TestSettings
import one.tesseract.crabdroid.RustObject

class TestSettingsProvider(handle: Long) : RustObject(handle) {
    external fun load(): TestSettings
    external fun save(settings: TestSettings)
}

File: ios/Common/Rust/Settings/TestSettingsProvider.swift
//
//  TestSettingsProvider.swift
//  Developer Wallet
//
//  Created by Daniel Leping on 26/01/2023.
//

import Foundation
import TesseractUtils
import CWallet

protocol TestSettingsProvider {
    func load() throws -> TestSettings;
    func save(settings: TestSettings) throws;
}

extension SettingsProvider: TestSettingsProvider {
    func load() throws -> TestSettings {
        try CResult<CTestSettings>.wrap { value, error in
            wallet_test_settings_provider_load(self.rust, value, error)
        }.get().copied()
    }
    
    func save(settings: TestSettings) throws {
        try CResult<Void>.wrap { err  in
            wallet_test_settings_provider_save(self.rust, settings.copiedPtr(), err)
        }.get()
    }
}
